src/natcap/invest/recreation/recmodel_server.py

```diff
@@ -540,59 +827,96 @@ def _file_len(file_path):
     if wc_process.returncode != 0:
         LOGGER.warning(err)
         return -1
-    return int(result.strip().split()[0])
+    return int(result.strip().split()[-2]) * multiplier
 
 
 def construct_userday_quadtree(
-        initial_bounding_box, raw_photo_csv_table, cache_dir,
-        max_points_per_node):
+        initial_bounding_box, raw_csv_file_list, dataset_name, cache_dir,
+        ooc_qt_picklefilename, max_points_per_node, max_depth,
+        n_workers=None, build_shapefile=True, fast_point_count=False):
     """Construct a spatial quadtree for fast querying of userday points.
 
     Args:
         initial_bounding_box (list of int):
-        raw_photo_csv_table ():
+        raw_csv_file_list (list): list of filepaths of point CSVs
+        dataset_name (string): one of 'flickr', 'twitter', indicating the
+            expected structure of the csv.
         cache_dir (string): path to a directory that can be used to cache
             the quadtree files on disk
+        ooc_qt_picklefilename (string): name for the pickle file quadtree index
+            created in the cache_dir.
         max_points_per_node(int): maximum number of points to allow per node
-            of the quadree.  A larger amount will cause the quadtree to
+            of the quadtree.  A larger amount will cause the quadtree to
             subdivide.
+        max_depth (int): maximum depth of nodes in the quadtree.
+            Once reached, the leaf nodes will not subdivide,
+            even if max_points_per_node is exceeded.
+        n_workers (int): number of cores for multiprocessing.
+        build_shapefile (boolean): whether or not to create vector geometries
+            representing nodes of the quadtree.
+        fast_point_count (boolean): If False, count the number of lines in all
+            the csv files. If True, estimate total number of points by counting
+            a random sample of files.
 
     Returns:
         None
     """
-    LOGGER.info('hashing input file')
-    start_time = time.time()
-    LOGGER.info(raw_photo_csv_table)
-    csv_hash = _hashfile(raw_photo_csv_table, fast_hash=True)
-
-    ooc_qt_picklefilename = os.path.join(cache_dir, csv_hash + '.pickle')
-    if os.path.isfile(ooc_qt_picklefilename):
-        return ooc_qt_picklefilename
+    LOGGER.info('counting lines in input file')
+    total_lines = _file_len(raw_csv_file_list, estimate=fast_point_count)
+    LOGGER.info('%d lines', total_lines)
+
+    # On a single CPU, flushing to disk is the main bottleneck.
+    # For large trees (i.e. twitter) more than 80% of wall-time
```

Comment:
What is meant by wall-time?

Reply:
I just meant "real" time elapsed, like watching the clock on the wall 🕐 . As opposed to time-elapsed * n_CPUs. I guess it's redundant since I said I tested on a single CPU anyway.